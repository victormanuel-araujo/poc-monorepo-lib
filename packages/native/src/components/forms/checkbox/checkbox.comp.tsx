import React from 'react'

import { Text, Icon, CheckboxProps, View } from '@components'
import { getThemeColor, Theme } from '@theme'
import { Animated, Platform, StyleProp, TouchableWithoutFeedback } from 'react-native'
import { ThemeContext } from 'styled-components/native'

import {
  CheckboxContent,
  CheckboxShadow,
  CheckboxStyled,
  CheckboxView,
  IconCheckedView,
  IconCheckViewWrapper,
  IconWrapper,
  LabelWrapper,
} from './checkbox.styles'

/**
 * The Checkbox is simple component commonly used to approve or select something.
 * It controls checked state and change.
 * Also, it has an onChange animation, which floats the checkbox, showing its shadow.
 */
export const Checkbox: React.FC<CheckboxProps> = ({ fluid, checked: value, ...props }) => {
  const [checked, setChecked] = React.useState(value)
  const [animationChecked] = React.useState<Animated.Value>(new Animated.Value(0))

  const theme: Theme = React.useContext(ThemeContext)

  /**
   * This effect guarantee always when the `checked` prop undergo a change, the
   * local state will be updated and sync
   */
  React.useEffect(() => {
    setChecked(value)
  }, [value])

  /**
   * This effect, exec the animation to the checkbox container always when
   * the `checked` undergo a change
   */
  React.useEffect(() => {
    /**
     * Control the `checked animation`.
     * This animation control the floating of the checkbox element, showing its shadow
     */
    const toValue: number = checked ? 1 : 0
    const bounciness: number = checked ? 5 : 0
    Animated.spring(animationChecked, { toValue, bounciness, useNativeDriver: false }).start()
  }, [checked, animationChecked])

  /**
   * Used to handle with the toggle check and call the onChange function
   */
  const handleCheck = () => {
    const newValue = !checked
    if (!props.disabled) {
      setChecked(newValue)
      if (props.onChange) {
        props.onChange(newValue)
      }
    }
  }

  /* --- Interpolation to the checkbox floating animation --- */

  const offsetCheckboxX: Animated.AnimatedInterpolation = animationChecked.interpolate({
    inputRange: [0, 1],
    outputRange: [0, props.offset],
  })
  const offsetCheckboxY: Animated.AnimatedInterpolation = animationChecked.interpolate({
    inputRange: [0, 1],
    outputRange: [0, -props.offset],
  })

  const backgroundColor: Animated.AnimatedInterpolation = animationChecked.interpolate({
    inputRange: [0, 1],
    outputRange: [
      getThemeColor(props.uncheckedBackgroundColor, props.uncheckedBackgroundColorShade)({ theme }),
      getThemeColor(props.checkedBackgroundColor, props.checkedBackgroundColorShade)({ theme }),
    ],
  })
  const borderColor: Animated.AnimatedInterpolation = animationChecked.interpolate({
    inputRange: [0, 1],
    outputRange: [
      getThemeColor(props.uncheckedBorderColor, props.uncheckedBorderColorShade)({ theme }),
      getThemeColor(props.checkedBorderColor, props.checkedBorderColorShade)({ theme }),
    ],
  })
  const _labelColor: Animated.AnimatedInterpolation = animationChecked.interpolate({
    inputRange: [0, 1],
    outputRange: [
      getThemeColor(props.uncheckedLabelColor, props.uncheckedLabelColorShade)({ theme }),
      getThemeColor(props.checkedLabelColor, props.checkedLabelColorShade)({ theme }),
    ],
  })

  /**
   * Apply the style to the `Checkbox` element.
   * Use the interpolation values to animate the fluctuation and show the shadow
   *
   * OBS: the `any` StyleProp, is because the `transform` is incompatible with
   * the interpolation value
   */
  const styledCheckbox: StyleProp<any> = {
    transform: [{ translateX: offsetCheckboxX }, { translateY: offsetCheckboxY }],
    borderColor,
    backgroundColor,
  }

  /**
   * Apply an disabled style to the checkbox
   */
  const disabledStyle: StyleProp<any> = {
    backgroundColor: getThemeColor('gray', 'lighter')({ theme }),
  }

  /**
   * Apply the style to the `Text` element (checkbox label).
   * Use the interpolation values to animate it with color transition
   *
   * OBS: the `any` StyleProp, is because the `transform` is incompatible with
   * the interpolation value
   */
  const labelColor: StyleProp<any> = {
    color: _labelColor,
  }

  return (
    <CheckboxStyled width={fluid ? '100%' : 'auto'} disabled={props.disabled} accessibilityRole="checkbox">
      <TouchableWithoutFeedback onPress={handleCheck} testID="checkbox" {...props}>
        <CheckboxContent width={fluid ? '100%' : 'auto'}>
          <CheckboxShadow />
          <CheckboxView
            width={fluid ? '100%' : 'auto'}
            style={props.disabled ? disabledStyle : styledCheckbox}
            spacing={props.spacing}
          >
            <IconWrapper>
              <IconCheckedView testID="checked-icon" disabled={props.disabled}>
                {checked && (
                  <IconCheckViewWrapper style={{ opacity: animationChecked }}>
                    <Icon.Material shape="none" size={18} icon="done" />
                  </IconCheckViewWrapper>
                )}
              </IconCheckedView>
            </IconWrapper>
            <LabelWrapper
              width="auto"
              spacing="md"
              paddingTop={0}
              paddingRight={0}
              paddingBottom={0}
              disabled={props.disabled}
            >
              <Text style={labelColor} weight="medium">
                {props.label}
              </Text>
            </LabelWrapper>
          </CheckboxView>
        </CheckboxContent>
      </TouchableWithoutFeedback>
    </CheckboxStyled>
  )
}

Checkbox.defaultProps = {
  fluid: true,
  spacing: 'md',
  checkedBorderColor: 'black',
  uncheckedBorderColor: 'black',
  checkedBackgroundColor: 'secondary',
  uncheckedBackgroundColor: 'white',
  checkedLabelColor: 'white',
  uncheckedLabelColor: 'black',
  offset: Platform.OS === 'web' ? 4 : 7,
}
