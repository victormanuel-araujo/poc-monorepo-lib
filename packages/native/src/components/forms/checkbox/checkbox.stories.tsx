import React from 'react'

import { Checkbox, CheckboxProps } from '@components'
import { Meta, Story } from '@storybook/react'

export default {
  title: 'Forms/Checkbox',
  component: Checkbox,
} as Meta

const Template: Story<CheckboxProps> = (args) => <Checkbox {...args} />

export const Default = Template.bind({})
Default.args = {
  fluid: false,
  label: 'Lorem Ipsum',
}

export const Checked = Template.bind({})
Checked.args = {
  ...Default.args,
  checked: true,
}

export const Correct = Template.bind({})
Correct.args = {
  ...Default.args,
  checked: true,
  checkedBackgroundColor: 'primary',
  checkedLabelColor: 'black',
}

export const WrongAnswer = Template.bind({})
WrongAnswer.args = {
  ...Default.args,
  checked: true,
  checkedBackgroundColor: 'red',
  checkedLabelColor: 'black',
}

export const CorrectAnswer = Template.bind({})
CorrectAnswer.args = {
  ...Default.args,
  uncheckedBackgroundColor: 'primary',
}
