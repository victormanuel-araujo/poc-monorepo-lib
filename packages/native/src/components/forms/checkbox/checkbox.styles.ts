import { CheckboxProps, View } from '@components'
import { getThemeColor, ThemedProps, ThemePaddings } from '@theme'
import { Animated, Platform } from 'react-native'
import styled, { css } from 'styled-components/native'

type DisabledThemed = ThemedProps<{ disabled?: boolean }>

/**
 * Apply the input border Radius.
 * It is used in more of one place
 */
function borderRadius() {
  return css`
    border-radius: ${({ theme }) => theme.units.radius}px;
  `
}

/**
 * Apply a custom style to the checkbox when it is rendered in a browser
 * @param disabled - boolean
 */
function webStyle({ disabled }: ThemedProps<{ disabled?: boolean }>) {
  if (Platform.OS === 'web') {
    return css`
      cursor: ${() => (disabled ? 'not-allowed' : 'pointer')};
    `
  }
}

/**
 * Apply the padding in the CheckBoxView
 */
function padding() {
  return css`
    padding: ${({ theme, spacing }: ThemedProps<Partial<CheckboxProps>>) =>
      `${theme.paddings.sm}px ${theme.paddings[spacing]}px`};
  `
}

/**
 * Apply the opacity if the checkbox is disabled
 */
function disabledOpacity() {
  return css`
    opacity: ${({ disabled }: DisabledThemed) => (disabled ? 0.7 : 1)};
  `
}

/**
 * Fix the label alignment in mobile devices.
 */
function labelAlignment() {
  const value = Platform.OS === 'web' ? 0 : 2
  return css`
    transform: translateY(-${value}px);
  `
}

export const CheckboxStyled = styled(View)<{ disabled?: boolean }>`
  ${webStyle};
`

export const CheckboxContent = styled(View)`
  width: auto;
  position: relative;
  flex-direction: row;
  min-height: 56px;
  padding: 0;
`

export const CheckboxView = styled(View)<ThemedProps<{ spacing: keyof ThemePaddings }>>`
  flex-direction: row;
  align-items: center;
  border: 1px solid;

  ${padding}
  ${borderRadius};
`

export const IconWrapper = styled(View)`
  position: relative;
  width: auto;
`

export const IconCheckedView = styled(View)<DisabledThemed>`
  border-radius: 12px;
  width: 24px;
  height: 24px;
  border: 1px solid ${getThemeColor('black')};
  align-items: center;
  justify-content: center;
  background-color: ${getThemeColor('white')};
  ${disabledOpacity}
`

export const IconCheckViewWrapper = styled(Animated.View)`
  position: relative;
  display: flex;
  align-content: center;
  justify-content: center;
`

export const LabelWrapper = styled(View)<DisabledThemed & { fluid?: boolean }>`
  width: auto;
  justify-content: center;
  align-items: flex-start;
  ${disabledOpacity};
  ${labelAlignment};
`

export const CheckboxShadow = styled(View)<ThemedProps<Partial<CheckboxProps>>>`
  width: 100%;
  height: 100%;
  padding: 0;
  position: absolute;
  left: 0;
  top: 0;
  background-color: ${({ checkedBackgroundColor, checkedBackgroundColorShade }) =>
    getThemeColor(checkedBackgroundColor, checkedBackgroundColorShade)};
  ${borderRadius}
`
