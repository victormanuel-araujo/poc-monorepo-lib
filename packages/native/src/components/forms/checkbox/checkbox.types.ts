import { TextProps } from '@components'
import { ThemeColorsName, ThemeColorsShade, ThemePaddings } from '@theme'

export interface CheckboxProps {
  /**
   * The label of the checkbox.
   * The text label that is placed in the checkbox.
   *
   */
  label: TextProps['children']

  /**
   * Fluid define the width of the checkbox. If the fluid is set to true, the
   * checkbox will occupy 100% of the father component.
   *
   * @default true
   */
  fluid?: boolean

  /**
   * Spacing define the checkbox padding.
   *
   * @default "md"
   */
  spacing?: keyof ThemePaddings

  /**
   * Define the border color when checked.
   *
   * @default "black"
   */
  checkedBorderColor?: ThemeColorsName

  /**
   * Define the border color shade when the checkbox is checked.
   *
   * @default "base"
   */
  checkedBorderColorShade?: keyof ThemeColorsShade

  /**
   * Define the border color when the checkbox is not checked.
   *
   * @default "black"
   */
  uncheckedBorderColor?: ThemeColorsName

  /**
   * Define the border color shade when the checkbox is not checked.
   *
   * @default "base"
   */
  uncheckedBorderColorShade?: keyof ThemeColorsShade

  /**
   * Define the distance between the input and its shadow, the higher, the farther
   * the input will fluctuate
   *
   * @default 7
   */
  offset?: number

  /**
   * Define the color of the checkbox background when is checked, using the color names
   *
   * @default "secondary"
   */
  checkedBackgroundColor?: ThemeColorsName

  /**
   * Define the color of the checkbox background when is checked, using the color shade.
   *
   * @default "base"
   */
  checkedBackgroundColorShade?: keyof ThemeColorsShade

  /**
   * Define the color of the checkbox shadow when isn't checked, using the color names
   *
   * @default "white"
   */
  uncheckedBackgroundColor?: ThemeColorsName

  /**
   * Define the color of the checkbox shadow when isn't checked, using the theme color shade.
   *
   * @default "base"
   */
  uncheckedBackgroundColorShade?: keyof ThemeColorsShade

  /**
   * Define the color of the label checkbox when its checked, using theme colors name.
   *
   * @default "black"
   */
  checkedLabelColor?: ThemeColorsName

  /**
   * Define the color shade of the label checkbox when its checked, using theme colors shade.
   *
   * @default "base"
   */
  checkedLabelColorShade?: keyof ThemeColorsShade

  /**
   * Define the color of the label checkbox when isn't checked, using theme colors name.
   *
   * @default "black"
   */
  uncheckedLabelColor?: ThemeColorsName

  /**
   * Define the color of the label checkbox when isn't checked using theme colors shade.
   *
   * @default "base"
   */
  uncheckedLabelColorShade?: keyof ThemeColorsShade

  /**
   * The value of the checkbox.
   *
   * The value that is set in the checkbox.
   *
   * @default false
   */
  checked?: boolean

  /**
   * Enable and Disable the checkbox. If it is `true`, the value of the checkbox
   * can't be changed
   *
   * @default false
   */
  disabled?: boolean

  /**
   * Trigger an event whenever the user clicks the checkbox.
   *
   * The `checked` state will be true when the user clicks the checkbox and false if it clicks again.
   *
   * @default `undefined`
   */
  onChange?(newValue: boolean): void
}
