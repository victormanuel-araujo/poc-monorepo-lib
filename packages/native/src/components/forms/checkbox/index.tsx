export * from './checkbox.comp'
export * from './checkbox.types'
