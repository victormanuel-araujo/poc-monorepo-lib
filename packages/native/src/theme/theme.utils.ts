// utils functions to handle theme
import { get, defaultsDeep } from 'lodash'
import { Platform } from 'react-native'
import { css } from 'styled-components/native'

import { theme as _theme } from './theme'
import {
  Theme,
  ThemeColorsName,
  ThemeColorsShade,
  ThemeFontWeights,
  ThemeFontSizes,
  ThemePaddings,
  ThemeProps,
  ThemeFonts,
} from './theme.types'

/**
 * This function create an theme object.
 * Optionally, you can pass an custom theme object with custom values and it will
 * always be merged with the default theme (smash)
 *
 * @param theme - custom theme
 *
 * @example
 * ```ts
 * const theme_a = { a: 1, b: 2, c: 3 };
 * const theme_b = { a: 2 };
 *
 * const theme_c = createTheme(theme_a, theme_b);
 *
 * // the output (theme_c) is equal to:
 * // theme_c = { a: 2, b: 2, c: 3 };
 * ```
 */
export const createTheme = (theme?: DeepPartial<Theme>): Theme => {
  return defaultsDeep(theme, _theme)
}

/**
 * easy way to get an theme color with a shade and optionally an alpha channel
 *
 * @param colorName - the name of the color
 * @param colorShade - the shade of the color
 * @param alpha - an alpha value in hexadecimal pattern, that is: two character from '00' to 'FF'
 *
 * @example
 * ```ts
 * inside a styled-component
 * ${getThemeColor(...props)}
 *
 * outside a styled-component
 * getThemeColor(...props)({ theme })
 * ```
 */
export const getThemeColor =
  (colorName: ThemeColorsName, colorShade: keyof ThemeColorsShade = 'base', alpha?: string) =>
  ({ theme }): string => {
    const color = get(theme, ['colors', colorName], { base: '#000000' })
    const shade: string = get(color, [colorShade], color.base)

    return alpha && alpha.length === 2 ? `${shade}${alpha}` : shade
  }

/**
 * Get the style of a font weight.
 * Verify the platform where the component is used, and return te correct style
 * to apply the font weight
 *
 * @param weight - Font weight, ie light, regular, bold, etc.
 * @param family - Font family - only used on Android
 *
 * @example
 * ```ts
 * const Bold = styled.Text`
 *   ${getFontWeight('bold')}
 * `
 * ```
 */
export const getFontWeight =
  (weight: keyof ThemeFontWeights, family: keyof ThemeFonts = 'default') =>
  ({ theme }) => {
    /*
     * In android, is necessary to change the font family too, cause the
     * font-weight property not work correctly
     */
    if (Platform.OS === 'android') {
      /* get the font family based on "family" prop  */
      const fontFamily: string = get(theme, ['fonts', family])

      /* return the style, with font-family and font-weight defined correctly */
      return css`
        font-family: ${fontFamily}-${weight};
        font-weight: ${get(theme, ['fontWeights', weight])};
      `
    }

    /*
     * In another platforms, as IOS or Web, is not necessary to set a new font family
     * so,
     */
    return css`
      font-weight: ${theme.fontWeights[weight]};
    `
  }

/**
 * Get the correct style for a font size.
 * return a style, with the font size and the relative line-height.
 *
 * @param size - Font size name, ie sm, md, lg, etc.
 * @param rawValue - If it's true, returns the raw integer value
 *                   instead of a style.
 *
 * @example
 * ```ts
 * const Title = styled.Text`
 *   ${getFontSize('lg')};
 * `
 * ```
 *
 * @example - `theme` comes from styled-component's ThemeContext.
 * ```tsx
 *   return <Icon size={getFontSize('lg', true)({ theme })} />
 * ```
 */
export const getFontSize =
  (size: keyof ThemeFontSizes, rawValue = false) =>
  ({ theme }: ThemeProps<Theme>) => {
    if (rawValue) {
      return get(theme, ['fontSizes', size])
    }

    return css`
      font-size: ${get(theme, ['fontSizes', size])}px;
      line-height: ${get(theme, ['lineHeights', size])}px;
    `
  }

/**
 * Used to get a style of a font family and verify the platform where the
 * component is used, and return te correct style
 * to apply the font family
 *
 * @param family - the name of the family
 * @param weight - optionally an weight of the family. use it only in android
 *
 * *
 * @example
 * ```ts
 * const TitleFamily = styled.Text`
 *   ${getFontFamily('title')}
 * `
 * ```
 */
export const getFontFamily =
  (family: keyof ThemeFonts = 'default', weight: keyof ThemeFontWeights = 'medium') =>
  ({ theme }: ThemeProps<Theme>) => {
    /* get the font family based on "family" prop  */
    const fontFamily: string = get(theme, ['fonts', family])

    if (Platform.OS === 'android') {
      /* return the style, with font-family defined correctly */
      return css`
        font-family: ${fontFamily}-${weight};
      `
    }

    return css`
      font-family: ${fontFamily};
    `
  }

/**
 * Get a named padding from current theme.
 *
 * @param size - Padding size name, ie sm, md, lg, etc.
 * @param multiply - used if is necessary to multiply the padding value
 *
 * @example
 * ```ts
 * const Container = styled.View`
 *   padding: ${getPadding('md')} ${getPadding('lg')};
 * `
 * ```
 */
export const getPadding =
  (size: keyof ThemePaddings, multiply = 1) =>
  ({ theme }: ThemeProps<Theme>): string => {
    return theme.paddings[size] * multiply + 'px'
  }
