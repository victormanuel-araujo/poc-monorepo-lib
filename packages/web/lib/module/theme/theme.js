import { colors, fonts, fontSizes, fontWeights, lineHeights, paddings, units } from "./theme.values";
/*
 * in the future, if exists a second theme, just have to create a second
 * composition as same is `theme`.
 */

/**
 * Default gama theme - SMASH
 */

export const theme = {
  colors,
  fonts,
  fontSizes,
  fontWeights,
  lineHeights,
  units,
  paddings
};
//# sourceMappingURL=theme.js.map