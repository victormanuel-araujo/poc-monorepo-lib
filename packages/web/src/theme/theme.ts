import { Theme } from './theme.types'
import { colors, fonts, fontSizes, fontWeights, lineHeights, paddings, units } from './theme.values'
/*
 * in the future, if exists a second theme, just have to create a second
 * composition as same is `theme`.
 */

/**
 * Default gama theme - SMASH
 */
export const theme: Theme = {
  colors,
  fonts,
  fontSizes,
  fontWeights,
  lineHeights,
  units,
  paddings,
}
