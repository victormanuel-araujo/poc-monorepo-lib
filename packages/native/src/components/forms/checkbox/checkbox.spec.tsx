import React from 'react'

import { fireEvent } from '@testing-library/react-native'

import { renderWithTheme } from '../../../theme/theme.fixture'

import { Checkbox } from './checkbox.comp'

describe('Checkbox component', () => {
  it('should render correctly', () => {
    const component = renderWithTheme(<Checkbox label="Lorem ipsum" />)

    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should call the onChange after user press', () => {
    const fn = jest.fn()
    const component = renderWithTheme(<Checkbox label="Lorem ipsum" onChange={fn} />)
    const checkboxButton = component.getByTestId('checkbox')

    fireEvent.press(checkboxButton)

    expect(fn).toHaveBeenCalledWith(true)
  })

  it('should call the onChange after user clicks passing a checked value as true ', () => {
    const fn = jest.fn()
    const component = renderWithTheme(<Checkbox label="Lorem ipsum" onChange={fn} checked={true} />)
    const checkboxButton = component.getByTestId('checkbox')

    fireEvent.press(checkboxButton)

    expect(fn).toHaveBeenCalledWith(false)
  })

  it('should change the check icon after user clicks', () => {
    const component = renderWithTheme(<Checkbox label="Lorem ipsum" />)
    const checkboxButton = component.getByTestId('checkbox')

    fireEvent.press(checkboxButton)

    expect(component.getByTestId('checked-icon')).toBeDefined()
  })

  it('should render with a label if that was passed as a parameter', () => {
    const component = renderWithTheme(<Checkbox label={'Lorem ipsum'} />)

    expect(component.queryByText('Lorem ipsum')).toBeDefined()
  })

  it('should not trigger the onChange if the disable was passed as a parameter', () => {
    const fn = jest.fn()
    const component = renderWithTheme(<Checkbox label="Lorem ipsum" disabled onChange={fn} />)
    const checkboxButton = component.getByTestId('checkbox')

    fireEvent.press(checkboxButton)

    expect(fn).not.toHaveBeenCalled()
  })
})
