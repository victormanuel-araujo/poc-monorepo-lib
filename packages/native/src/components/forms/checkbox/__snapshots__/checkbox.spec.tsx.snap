// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component should render correctly 1`] = `
<View
  accessibilityRole="checkbox"
  backgroundShade="base"
  dir="column"
  display="flex"
  position="relative"
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "paddingBottom": 0,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 0,
      "position": "relative",
      "width": "100%",
    }
  }
  width="100%"
>
  <View
    accessible={true}
    backgroundShade="base"
    dir="column"
    display="flex"
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    position="relative"
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "minHeight": 56,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
        "position": "relative",
        "width": "auto",
      }
    }
    testID="checkbox"
    width="100%"
  >
    <View
      backgroundShade="base"
      dir="column"
      display="flex"
      position="relative"
      style={
        Object {
          "backgroundColor": "#000000",
          "borderRadius": 8,
          "display": "flex",
          "flexDirection": "column",
          "height": "100%",
          "left": 0,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        }
      }
    />
    <View
      backgroundShade="base"
      dir="column"
      display="flex"
      position="relative"
      spacing="md"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(255, 255, 255, 1)",
          "borderColor": "rgba(0, 0, 0, 1)",
          "borderRadius": 8,
          "borderStyle": "solid",
          "borderWidth": 1,
          "display": "flex",
          "flexDirection": "row",
          "paddingBottom": 8,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 8,
          "position": "relative",
          "transform": Array [
            Object {
              "translateX": 0,
            },
            Object {
              "translateY": 0,
            },
          ],
          "width": "100%",
        }
      }
      width="100%"
    >
      <View
        backgroundShade="base"
        dir="column"
        display="flex"
        position="relative"
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
            "position": "relative",
            "width": "auto",
          }
        }
      >
        <View
          backgroundShade="base"
          dir="column"
          display="flex"
          position="relative"
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderColor": "#000000",
              "borderRadius": 12,
              "borderStyle": "solid",
              "borderWidth": 1,
              "display": "flex",
              "flexDirection": "column",
              "height": 24,
              "justifyContent": "center",
              "opacity": 1,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
              "position": "relative",
              "width": 24,
            }
          }
          testID="checked-icon"
        />
      </View>
      <View
        backgroundShade="base"
        dir="column"
        display="flex"
        paddingBottom={0}
        paddingRight={0}
        paddingTop={0}
        position="relative"
        spacing="md"
        style={
          Object {
            "alignItems": "flex-start",
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 0,
            "paddingLeft": 16,
            "paddingRight": 0,
            "paddingTop": 0,
            "position": "relative",
            "transform": Array [
              Object {
                "translateY": -2,
              },
            ],
            "width": "auto",
          }
        }
        width="auto"
      >
        <Text
          accessibilityRole="text"
          style={
            Object {
              "color": "rgba(0, 0, 0, 1)",
              "fontFamily": "Raleway",
              "fontSize": 16,
              "fontWeight": "500",
              "lineHeight": 24,
            }
          }
          type="description"
          weight="medium"
        >
          Lorem ipsum
        </Text>
      </View>
    </View>
  </View>
</View>
`;
